packages/jsonapi/middleware.js

```diff
@@ -250,7 +251,14 @@ class Handler {
   }
 
   async handleCollectionGET(type) {
-    let { data: models, meta: { page }, included } = await this.searcher.search(this.session, this.branch, {
+    let branch = this.branch;
+
+    if(type === 'branches') {
```

Comment:
Putting a special case here is unlikely to be correct, because jsonapi is not the only place somebody might ask about branches.

If this special case is really needed, we should move it down into `hub/searchers.js`. Then it will also work for (for example) a plugin that is calling the searchers directly, or for a future GraphQL middleware plugin.

Reply:
I have removed this special case and "moved it one layer down" but I think we have opened up a different set of unintended consequences as a result 🙈 I'll comment on the issue so I can bring it to your attention 👍